Chore: update mcd-seth tutorial

hbarcelos opened the pull request:
- [x] Replace all addresses to use the latest Goerli environment instead of Kovan.
- [x] Replace `BAT-A` by `WBTC-A` as the ilk used for the examples.
- [x] Streamline some `seth` calls, getting rid of redundant/unnecessary calls.
- [x] Fix some `seth --to-dec` calls causing problems when the data was already in the decimal format.
- [x] Improve general markdown structure.
- [x] Fix some typos.

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -1,20 +1,23 @@
-  - [Paying back DAI debt to release collateral](#paying-back-dai-debt-to-release-collateral)
+**This guide works under the Goerli [1.11.0 Release](https://changelog.makerdao.com/releases/goerli/1.11.0/index.html) of the system.**
 
 **You can also use the Rinkey, Ropsten and Goerli deployments in this guide. Just make sure to change the contract addresses from the specific network.**
```
```suggestion
**You can also use the Rinkey, Ropsten and Kovan deployments in this guide. Just make sure to change the contract addresses from the specific network.**
```

naszam replied:
```suggestion
**You can also use the Rinkey, Ropsten and Kovan deployments in this guide. Just make sure to change the contract addresses from the specific network.**
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -1,20 +1,23 @@
-This tutorial will cover how to use the tool `seth` to deposit BAT tokens to draw DAI from the Kovan deployment of MCD as an example, since the process is the same for any other ERC-20 token. You can use the same methodology for any supported token in MCD, by changing the contract addresses to the specific token you want to use.
+This tutorial will cover how to use the tool `seth` to deposit WBTC tokens to draw DAI from the Goerli deployment of MCD as an example, since the process is the same for any other ERC-20 token. You can use the same methodology for any supported token in MCD, by changing the contract addresses to the specific token you want to use.
+
+<!-- vim-markdown-toc GFM -->
```
It seems that some vim artifacts were introduced alongside the changes. I would suggest to cleanup those.
```suggestion
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -1,20 +1,23 @@
+* [Finally interacting with the Maker Protocol contracts](#finally-interacting-with-the-maker-protocol-contracts)
+* [Paying back DAI debt to release collateral](#paying-back-dai-debt-to-release-collateral)
+
+<!-- vim-markdown-toc -->
```
Same as above.
```suggestion
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -133,184 +170,243 @@ In order to better understand the MCD contracts, the following provides a brief
 
-The contract function looks like the following: `join(address urn, uint wad)`
+First we are going to make a transaction to the WBTC adapter to actually take 5 of our tokens with the join contract function.
+The contract function looks like the following: `join(address urn, uint256 wad)`
```
I'd suggest to maybe add a note to specify that the WBTC adapter uses a non standard GemJoin ([GemJoin5](https://github.com/makerdao/dss-gem-joins/blob/master/src/join-5.sol)).
```suggestion
The contract function looks like the following: `join(address urn, uint256 amt)`
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -133,184 +170,243 @@ In order to better understand the MCD contracts, the following provides a brief
+The contract function looks like the following: `join(address urn, uint256 wad)`
 
 - The first parameter is the `urn`, our vault address.
 - The second parameter is the token amount in `wad`.
```
```suggestion
- The second parameter is the token amount `amt`.
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
````diff
@@ -133,184 +170,243 @@ In order to better understand the MCD contracts, the following provides a brief
+```
 
-`export wadC=$(seth --to-uint256 $(seth --to-wei 150 eth))`
+Then use the following command to use the join function, thus taking 150 WBTC from our account and sending to `urn` address.
````
```suggestion
Then use the following command to use the join function, thus taking 5 WBTC from our account and sending to `urn` address.
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
````diff
@@ -133,184 +170,243 @@ In order to better understand the MCD contracts, the following provides a brief
+```
 
-`seth send $MCD_JOIN_BAT_A "join(address, uint)" $urn $wadC`
+**NOTICE:** From this point on, the join adapter already took care of the fact that WBTC has only 8 decimals, so we can proceed with `wad` normally.
````
```suggestion
**NOTICE:** From this point on, the join-5 adapter already took care of the fact that WBTC has only 8 decimals, so we can proceed with `wad` normally.
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -133,184 +170,243 @@ In order to better understand the MCD contracts, the following provides a brief
+If the `frob` operation is successful, it will adjust the corresponding data in the protected `vat` module. When adding collateral to an `urn`, `dink` needs to be the (positive) amount we want to add and `dart` needs to be the (positive) amount of DAI we want to draw. 
 
-We already set up `cdp` before, so we only need to set up `dink` (BAT deposit) and `dart` (DAI to be drawn):
+Let’s add our 150 WBTC to the urn, and draw 15000 DAI ensuring that the position is overcollateralized.
```
```suggestion
Let’s add our 5 WBTC to the urn, and draw 15000 DAI ensuring that the position is overcollateralized.
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -133,184 +170,243 @@ In order to better understand the MCD contracts, the following provides a brief
-dink=$(seth --to-uint256 $(seth --to-wei 150 eth))
-rate=$(seth --to-fix 27 $(seth --to-dec $(seth call $MCD_VAT 'ilks(bytes32)(uint256,uint256,uint256,uint256,uint256)' $ilk | sed -n 2p)))
-dart=$(seth --to-uint256 $(bc<<<"scale=18; art=(20/$rate*10^18+1); scale=0; art/1"))
+dink=$(seth --to-wei 5 eth)
+rate=$(seth call $MCD_VAT 'ilks(bytes32)(uint256 Art, uint256 rate, uint256 spot, uint256 line, uint256 dust)' $ilk | \
+    sed -n 2p | seth --to-fix 27)
+dart=$(bc<<<"scale=18; art=(15000/$rate*10^18+1); scale=0; art/1")
```
```suggestion
export dink=$(seth --to-wei 5 eth)
export rate=$(seth call $MCD_VAT 'ilks(bytes32)(uint256 Art, uint256 rate, uint256 spot, uint256 line, uint256 dust)' $ilk | \
    sed -n 2p | seth --to-fix 27)
export dart=$(bc<<<"scale=18; art=(15000/$rate*10^18+1); scale=0; art/1")
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -125,192 +168,274 @@ In order to better understand the MCD contracts, the following provides a brief
-dink=$(seth --to-uint256 $(seth --to-wei 150 eth))
-rate=$(seth --to-fix 27 $(seth --to-dec $(seth call $MCD_VAT 'ilks(bytes32)(uint256,uint256,uint256,uint256,uint256)' $ilk | sed -n 2p)))
-dart=$(seth --to-uint256 $(bc<<<"scale=18; art=(20/$rate*10^18+1); scale=0; art/1"))
+export RAD_DECIMALS=45
+export RAY_DECIMALS=27
+export WAD_DECIMALS=18
```
```suggestion
export WAD_DECIMALS=18
export RAY_DECIMALS=27
export RAD_DECIMALS=45
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -125,192 +168,274 @@ In order to better understand the MCD contracts, the following provides a brief
-rate=$(seth --to-fix 27 $(seth --to-dec $(seth call $MCD_VAT 'ilks(bytes32)(uint256,uint256,uint256,uint256,uint256)' $ilk | sed -n 2p)))
-debt=$(bc<<<"$art*$rate")
-debtWadRound=$(seth --to-uint256 $(bc<<<"$art*$rate*10^18/1+1"))
+export RAD_DECIMALS=45
+export RAY_DECIMALS=27
+export WAD_DECIMALS=18
```
Same as above.
```suggestion
export WAD_DECIMALS=18
export RAY_DECIMALS=27
export RAD_DECIMALS=45
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
````diff
@@ -125,192 +168,274 @@ In order to better understand the MCD contracts, the following provides a brief
 
 ```bash
-dink=$(seth --to-uint256 $(mcd --to-hex $(seth --to-wei -150 eth)))
+dink=$(seth --to-uint256 $(mcd --to-hex $(seth --to-wei -5 eth)))
````
I'd suggest to consider the following change as It seems that at the time of writing of this tutorial `seth` couldn't manage negative integers, that now seem to be supported.
The following should return the same output:
```suggestion
dink=$(seth --to-int256 $(seth --to-wei -5 eth))
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
````diff
@@ -125,192 +168,274 @@ In order to better understand the MCD contracts, the following provides a brief
 ```bash
-dink=$(seth --to-uint256 $(mcd --to-hex $(seth --to-wei -150 eth)))
+dink=$(seth --to-uint256 $(mcd --to-hex $(seth --to-wei -5 eth)))
 dart=$(seth --to-uint256 $(mcd --to-hex -$(seth --to-wei $art eth)))
````
Same as above.
```suggestion
dart=$(seth --to-int256 -$(seth --to-wei $art eth))
```

hbarcelos replied:
Does this mean we can we get rid of the `mcd-cli` dependency?

naszam replied:
yep

naszam commented on mcd/mcd-seth/mcd-seth.md:
````diff
@@ -125,192 +168,274 @@ In order to better understand the MCD contracts, the following provides a brief
+
+The WBTC is still assigned to the Vault, so we need to move them to our address:
+```bash
+export wadC=$(seth --to-wei 5 eth)
````
I'd suggest to rename it `wad`.
```suggestion
export wad=$(seth --to-wei 5 eth)
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
````diff
@@ -125,192 +168,274 @@ In order to better understand the MCD contracts, the following provides a brief
+The WBTC is still assigned to the Vault, so we need to move them to our address:
+```bash
+export wadC=$(seth --to-wei 5 eth)
+seth send $CDP_MANAGER 'flux(uint256, address, uint256)' $cdpId $ETH_FROM $wadC
````
Same as above.
```suggestion
seth send $CDP_MANAGER 'flux(uint256, address, uint256)' $cdpId $ETH_FROM $wad
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
````diff
@@ -125,192 +168,274 @@ In order to better understand the MCD contracts, the following provides a brief
 
-Then use the following command to use the join function, thus taking 150 BAT from our account and sending to `urn` address.
+```bash
+seth send $MCD_JOIN_WBTC_A "join(address, uint256)" $urn $amt
````
```suggestion
seth send $MCD_JOIN_WBTC_A 'join(address, uint256)' $urn $amt
```

naszam replied:
```suggestion
seth send $MCD_JOIN_WBTC_A 'join(address, uint256)' $urn $amt
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
````diff
@@ -125,192 +168,274 @@ In order to better understand the MCD contracts, the following provides a brief
 
-`seth send $MCD_JOIN_DAI "join(address,uint)" $urn $debtWadRound`
+```bash
+seth send $MCD_JOIN_DAI "join(address, uint256)" $urn $debtWadRound
````
```suggestion
seth send $MCD_JOIN_DAI 'join(address, uint256)' $urn $debtWadRound
```
```suggestion
seth send $MCD_JOIN_DAI "join(address, uint256)" $urn $debtWadRound
```

naszam replied:
```suggestion
seth send $MCD_JOIN_DAI 'join(address, uint256)' $urn $debtWadRound
```
```suggestion
seth send $MCD_JOIN_DAI "join(address, uint256)" $urn $debtWadRound
```

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -125,192 +168,274 @@ In order to better understand the MCD contracts, the following provides a brief
 
-`seth send $MCD_JOIN_BAT_A "exit(address, uint)" $ETH_FROM $wadC`
+export amt=$(seth --from-fix $WBTC_DECIMALS 5)
+seth send $MCD_JOIN_WBTC_A "exit(address, uint256)" $ETH_FROM $amt
```
```suggestion
seth send $MCD_JOIN_WBTC_A 'exit(address, uint256)' $ETH_FROM $amt
```

naszam replied:
```suggestion
seth send $MCD_JOIN_WBTC_A 'exit(address, uint256)' $ETH_FROM $amt
```

hbarcelos commented:
I carefully reviewed all quoting and made some other improvements.  
Should be good to go now.

naszam commented on mcd/mcd-seth/mcd-seth.md:
````diff
@@ -125,192 +167,273 @@ In order to better understand the MCD contracts, the following provides a brief
-`seth send $CDP_MANAGER 'open(bytes32,address)' $ilk $ETH_FROM`
+We begin by opening an empty Vault, so we can use it to lock collateral into. For this we need to define the type of collateral (WBTC-A) we want to lock in this Vault:
+```bash
+export ilk=$(seth --to-bytes32 $(seth --from-ascii 'WBTC-A'))
````
I'd keep the double quotes here.
```suggestion
export ilk=$(seth --to-bytes32 $(seth --from-ascii "WBTC-A"))
```

naszam replied:
see: https://collateral.makerdao.com/smart-contracts-domain/untitled

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -125,192 +167,273 @@ In order to better understand the MCD contracts, the following provides a brief
+
+Inside the `vat`, different parameters have different decimal precisions:
+
+- `dai`: 45 decimals (`rad`)
+- `rate`: 27 decimals (`ray`)
+- `dink`/`dart`: 18 decimals (`wad`).
```
```suggestion
- `dink`: 18 decimals (`wad`)
- `dart`: 18 decimals (`wad`)
- `rate`: 27 decimals (`ray`)
- `dai`: 45 decimals (`rad`)
```

hbarcelos replied:
Not sure if this should be sorted by the number of decimals. I made it match the usage order in the article.

naszam commented on mcd/mcd-seth/mcd-seth.md:
```diff
@@ -125,192 +167,273 @@ In order to better understand the MCD contracts, the following provides a brief
+The function is called `frob(uint256, uint256, uint256)`, which receives the following parameters:
+
+- `uint256 cdp`: the `cdpId`
+- `int256 dink`: delta ink (collateral)
+- `int256 dart`: delta art (Dai).
```
I'd suggest the following as alternative to the above.
```suggestion
- `int256 dink`: delta ink (collateral) `[wad]`
- `int256 dart`: delta art (Dai) `[wad]`
```

hbarcelos replied:
I think it's better if we do both to add some context.

hbarcelos commented:
I made some final adjustments and addressed all your feedback.

I'd suggest that fork the sake of the time of us both we refrain from further back and forth in this PR unless there's something critically wrong with it.

lollike reviewed (approved):
LGTM - will fix a small typo myself

Dispatch700 commented:
Thank you for your email. I am Juan with Dispatch Connections and I would
like to schedule a meeting with you to discuss the opportunity to increase
the acquisition of clients worldwide that utilize your program. Through a
specifically designed campaign that promotes your organization through a
combined advertisement & marketing campaign that provides promotions for
your organization. The great thing about this opportunity is that you only
pay for the clients that sign up to use your products and services.
Through a 30 minute strategy call we will analyze your current client base
attrition versus acquisition rate and determine the expected results in the
investment and the expected ROI based on that investment. ROI (Returns on
Investment) - That's right this is not a cost that your company must incur
for marketing of your product but yet an investment that will bring returns
beyond just the acquired clients but also gains on principal and guaranteed
returns. Let's schedule a consultation to discuss how we can work together.
Use the link here to schedule a Free consultation HERE
<https://calendly.com/jpcountry501c3/15min?month=2022-04>.

We are excited to offer businesses such as yours the opportunity to benefit
from the innovative blockchain market that allows for these unusual gains
through principal investments in these crypto based marketing campaigns. It
offers transparency, accountability, and most important returns on
investment. Learn more about the marketing strategy HERE
<https://jpcinvestorsportal.wixsite.com/website>

On Fri, Apr 8, 2022 at 8:02 AM Nikolaj Lollike ***@***.***>
wrote:

> Merged #98 <https://github.com/makerdao/developerguides/pull/98> into
> master.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/makerdao/developerguides/pull/98#event-6396300860>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AX7LIOX2RDR72DRZWKCLCC3VEAU5HANCNFSM5SFJ3NIQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
